### PR TITLE
add email to plan collaborator and add email domain to DB

### PR DESCRIPTION
**Pull request by StevenWadeOddball**
## Changes and Description

- Add Email address to Plan Collaborator
- Add Email domain to the db to enforce valid email address/

## How to test this change
1. scripts/dev up:backend:debug && scripts/dev db:seed 
2. Try to make a collaborator with a bad email address
```gql
mutation createPlanCollaborator {
  createPlanCollaborator(input: {
      
    modelPlanID: "6e224030-09d5-46f7-ad04-4bb851b36eab"
    euaUserID: "BC0V"
    fullName:"Clay B"

    teamRole: LEADERSHIP
    email: "ahsdklashdlaksd"
    }) {
    id
    modelPlanID
    euaUserID
    fullName
    email
    teamRole
  }
}

```
3. Ensure the above failed
4.  Make a collaborator with a valid email address
```gql
mutation createPlanCollaborator {
  createPlanCollaborator(input: {
      
    modelPlanID: "6e224030-09d5-46f7-ad04-4bb851b36eab"
    euaUserID: "BC0V"
    fullName:"Clay B"

    teamRole: LEADERSHIP
    email: "clay@oddball.com"
    }) {
    id
    modelPlanID
    euaUserID
    fullName
    email
    teamRole
  }
}

```

## NOTE

This change likely breaks the front end. The add collaborator logic will need to be updated to pass email as required to create collaborator.

## PR Author Review Checklist

- [ ] Met the ticket's acceptance criteria, or will meet them in a subsequent PR.
- [ ] Added or updated tests for backend resolvers or other functions as needed.
- [ ] Added or updated client tests for new components, parent components, functions, or e2e tests as necessary.
- [ ] Tested user-facing changes with voice-over and the [rotor menu](https://support.apple.com/guide/voiceover/with-the-voiceover-rotor-mchlp2719/mac)

## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.

### CODE REVIEW

File: pkg/models/plan_collaborator.go [hunk 1]
```diff
@@ -7,6 +7,7 @@ type PlanCollaborator struct {
 	EUAUserID string   `json:"euaUserId" db:"eua_user_id"`
 	FullName  string   `json:"fullName" db:"full_name"`
 	TeamRole  TeamRole `json:"teamRole" db:"team_role"`
+	Email     string   `json:"email" db:"email"`
```
Comment (ClayBenson94) [inline pkg/models/plan_collaborator.go:10]: Would you mind leaving a comment (either on the struct as a whole, or on the `FullName` and `Email` fields) describing that these are stored as a result from the initial CEDAR query that's made when we create the collaborator (effectively caching these values, which are unlikely to change)?

**Review by ClayBenson94 — CHANGES_REQUESTED** [review]

**Comment by patrickseguraoddball** [comment]
I added the FE email field to collaborator queries and mutations.  Works great

**Comment by ClayBenson94** [comment]
@patrickseguraoddball Looks like there might be a [failing snapshot](https://github.com/CMSgov/mint-app/runs/8093429115?check_suite_focus=true#step:6:566) - does that seem related to this change? 🤔

**Review by ClayBenson94 — DISMISSED** [review]

File: pkg/models/plan_collaborator.go (hunk 1 above)
Comment (ClayBenson94) [reply pkg/models/plan_collaborator.go:12]: b39e78894448f43c13dfa5e6f99d71763100b7b7

**Pull request merged**